File: Python3/프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    for garo in range(1, brown+yellow+1):
        if (brown+yellow) % garo == 0:
            # yellow = (garo-2) * (sero-2)
            # brown = (garo*2) + (sero-2)*2
            # yellow + brown = garo * sero
            sero = (brown+yellow) // garo
            
            if (garo-2) * (sero-2) == yellow:
                return [sero, garo]

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    int sum = brown + yellow;
    for(int w=1; w<=sum; w++) {
        for(int h=1; h<=sum; h++) {
            if(w * h != sum) {
                continue;
            }
            if(w < h) {
                continue;
            }
            if(w < 3 || h < 3) {
                continue;
            }
            
            if((w - 2) * (h - 2) == yellow) {
                answer.push_back(w);
                answer.push_back(h);
                return answer;
            }
        }
    }
}